### PR TITLE
add results limit, default 25

### PR DESCRIPTION
**Pull request by arodu**

### CODE REVIEW

**Comment by othercorey** [comment]
@ndm2

**Comment by ndm2** [comment]
Are you planning on adding a UI control to the theme that would allow users to easily change the limit? I'm pretty sure mostly nobody is going to add the `limit` query parameter to the URL manually, just on the off chance.

**Comment by arodu** [comment]
> Are you planning on adding a UI control to the theme that would allow users to easily change the limit? I'm pretty sure mostly nobody is going to add the `limit` query parameter to the URL manually, just on the off chance.

I made a console interface, and 25 results don't look good
https://github.com/CakeDC/Book/tree/2.next-cake4

File: src/Model/Index/SearchIndex.php [hunk 1]
```diff
@@ -32,7 +33,7 @@ public function search($lang, $version, $options = [])
         $this->setName($indexName);
         $query = $this->query();
 
-        $query->page($options['page'], 25)
+        $query->page($options['page'], $options['limit'])
```
Comment (ndm2) [inline src/Model/Index/SearchIndex.php:36]: Can we clamp that between 1 and a max limit?

File: src/Model/Index/SearchIndex.php [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 class SearchIndex extends Index
 {
+    protected const MAX_LIMIT = 25;
```
Comment (ADmad) [inline src/Model/Index/SearchIndex.php:9]: Clamping max limit to default limit seems quite limiting 🙂.

Comment (arodu) [reply src/Model/Index/SearchIndex.php:9]: I thought about not exceeding the limit that you already had before

so.. how do you want it?

Comment (ADmad) [reply src/Model/Index/SearchIndex.php:9]: Meh.. 25 is fine. We can see later if a higher value is actually necessary.

**Comment by arodu** [comment]
separate the default_limit from the max_limit, now you can change those values later

**Review by ADmad — APPROVED** [review]

**Comment by ndm2** [comment]
Sorry, forgot to notify you, but you probably saw that the overhauled search includes support for `limit` too.

The results are structured differently now too, so you'll have to change your console app a bit. There's no `title` field anymore, instead you can use the last entry in the `hierarchy` array, this will be the title of the matched section. Also `contents` isn't an array anymore (except for in the highlights), but a string.

By default matches are highlighted now, and unless configured otherwise, use double curly braces, which you could transform into terminal color codes for the CLI if you wanted to.

**Pull request closed without merging**